### PR TITLE
fix, docs:  use formattedName instead of name in getAllGroups method and add README.md

### PR DESCRIPTION
**Pull request by Vyogami**
This pull request includes multiple commits with the following changes:

- Adds IDE/Editor and dependency specific files to .gitignore, further improving version control. 
- Updates filtering to use contact formatted name instead of group, improving functionality and fixing a bug in which name was inaccessible from group object.
- Adds a README.md file, improving project documentation.
- Adds MIT License

### CODE REVIEW

**Comment by legitShivam** [comment]
@ditsuke does this looks good?

File: README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+await printMatchingGroups(client, 'ALiAS'); // replace ALiAS with your keyword
+```
+
+This will print the names and IDs of all groups whose names contain the word "family".
+
+Note that this program adds members to the target group in batches of 10 with a 5-second delay between batches to avoid being blocked by WhatsApp. If any errors occur while adding members, they will be logged to the console.
+
+## License
+
+This project is licensed under the MIT License. See the `LICENSE` file for details.
```
Comment (ditsuke) [inline README.md:37]: We don't have a LICENSE file yet. Can you include one with this PR?

**Review by ditsuke — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Vyogami) [reply README.md:37]: done, is this fine?

Comment (ditsuke) [reply README.md:37]: Looks good

**Review by ditsuke — APPROVED** [review]

**Comment by ditsuke** [comment]
Thanks!

**Pull request merged**